Assets/UI/launchbar_CQUI.lua

```diff
@@ -0,0 +1,250 @@
+  function OnToggleExtras()
+    if Controls.LaunchExtraShow:IsChecked() then
+      Controls.LaunchExtraControls:SetHide(true);
+  
+      Controls.LaunchExtraAlpha:SetToBeginning();
+      Controls.LaunchExtraSlide:SetToBeginning();
+  
+      Controls.LaunchExtraAlpha:Play();
+      Controls.LaunchExtraSlide:Play();
+  
+      Controls.LaunchExtraControls:SetHide(false);
+  
+      BuildExtraEntries();
+    else
+      OnCloseExtras()
+    end
+  end
+
+-- ===========================================================================
+function OnAddExtraEntry(entryKey:string, entryInfo:table)
+  -- Add info at key. Overwrite if they key already exists.
+  m_LaunchbarExtras[entryKey] = entryInfo
+end
+
+-- ===========================================================================
+function OnAddLaunchbarIcon(buttonInfo:table)
+  local tButtonEntry:table = {};
+  ContextPtr:BuildInstanceForControl("LaunchbarButtonInstance", tButtonEntry, Controls.ButtonStack);
+
+  local textureOffsetX = buttonInfo.IconTexture.OffsetX;
+  local textureOffsetY = buttonInfo.IconTexture.OffsetY;
+  local textureSheet = buttonInfo.IconTexture.Sheet;
+
+  -- Update Icon Info
+  if (textureOffsetX ~= nil and textureOffsetY ~= nil and textureSheet ~= nil) then
+    tButtonEntry.Image:SetTexture(textureOffsetX, textureOffsetY, textureSheet);
+  end
+  if (buttonInfo.IconTexture.Color ~= nil) then
+    tButtonEntry.Image:SetColor(buttonInfo.IconTexture.Color);
+  end
+
+  if (buttonInfo.Tooltip ~= nil) then
+    tButtonEntry.Button:SetToolTipString(buttonInfo.Tooltip);
+  end
+
+  textureOffsetX = buttonInfo.BaseTexture.OffsetX;
+  textureOffsetY = buttonInfo.BaseTexture.OffsetY;
+  textureSheet = buttonInfo.BaseTexture.Sheet;
+
+  local stateOffsetX = buttonInfo.BaseTexture.HoverOffsetX;
+  local stateOffsetY = buttonInfo.BaseTexture.HoverOffsetY;
+
+  if (textureOffsetX ~= nil and textureOffsetY ~= nil and textureSheet ~= nil) then
+    tButtonEntry.Base:SetTexture(textureOffsetX, textureOffsetY, textureSheet);
+    if (buttonInfo.BaseTexture.Color ~= nil) then
+      tButtonEntry.Base:SetColor(buttonInfo.BaseTexture.Color);
+    end
+
+    -- Setup behaviour on hover
+    if (stateOffsetX ~= nil and stateOffsetY ~= nil) then
+      local OnMouseOver = function()
+        tButtonEntry.Base:SetTextureOffsetVal(stateOffsetX, stateOffsetY);
+        UI.PlaySound("Main_Menu_Mouse_Over");
+      end
+
+      local OnMouseExit = function()
+        tButtonEntry.Base:SetTextureOffsetVal(textureOffsetX, textureOffsetY);
+      end
+
+      tButtonEntry.Button:RegisterMouseEnterCallback( OnMouseOver );
+      tButtonEntry.Button:RegisterMouseExitCallback( OnMouseExit );
+    end
+  end
+
+  if (buttonInfo.Callback ~= nil) then
+    tButtonEntry.Button:RegisterCallback( Mouse.eLClick, buttonInfo.Callback );
+  end
+
+  RefreshView();
+end
+
+-- ===========================================================================
+function TestLaunchBarExtension()
```

Comment:
Looks like a function for experiments. I'm just adding this note as a reminder to remove before merging to master.

Reply:
it was in the previous set of code - commented out.   I added it as a function, but this function is never called.

This is the LaunchBar thing that we don't use at the moment - I figured can do this change and then remove it later if we want to remove it entirely, so we at least have the history.

Reply:
added a comment, left the function